pkg/apiserver/filters/impersonation.go

```diff
@@ -123,6 +123,15 @@ func WithImpersonation(handler http.Handler, requestContextMapper api.RequestCon
 		oldUser, _ := api.UserFrom(ctx)
 		httplog.LogOf(req, w).Addf("%v is acting as %v", oldUser, newUser)
 
+		// clear all the impersonation headers from the request
+		req.Header.Del(authenticationapi.ImpersonateUserHeader)
```

Comment:
Front proxy isn't the same as impersonation. The impersonation headers should not change, or we'll be unable to write clients (like kubectl) that can confidently ask to do something as someone else. 

I would not expect the client cert used by a front proxy that is verified in order to accept user info from headers to have any API permissions associated with it.